Changing the Dimensions and BaseUnit for the FuelEfficiency

lipchev opened the pull request:
Changing the `Dimensions` and `BaseUnit` for the `FuelEfficiency`:
- `BaseUnit` changed from `LiterPer100Kilometers` (defaults to `+Infinity`) to `KilometersPerLiter` (defaults to `0`)
- added `Dimensions` of `"L": -2` (was previously treated as a dimensionless quantity)
- added the `MeterPerCubicMeter` (m/m³) unit
- updated the relevant tests

lipchev commented on Common/UnitDefinitions/FuelEfficiency.json:
```diff
@@ -1,55 +1,84 @@
+        }
+      ]
+    },
+    {
+      "SingularName": "MeterPerCubicMeter",
+      "PluralName": "MetersPerCubicMeter",
+      "BaseUnits": {
+        "L": "Meter"
+      },
+      "FromUnitToBaseFunc": "{x} / 1e6",
+      "FromBaseToUnitFunc": "{x} * 1e6",
+      "Localization": [
+        {
+          "Culture": "en-US",
+          "Abbreviations": [
+            "m/m³"
+          ]
+        }
+      ]
+    }
```
I only added this because of the `UnitSystem.SI` coverage, but I'm not really _set_ on having it: you decide.

lipchev commented on UnitsNet.Tests/CustomCode/FuelEfficiencyTests.cs:
```diff
@@ -17,25 +17,22 @@
+        protected override double MilesPerUkGallonInOneKilometerPerLiter => 2.824809363318222;
+        protected override double MilesPerUsGallonInOneKilometerPerLiter => 2.352145833333333;
+
+        [Fact(Skip = "See about changing this to MetersPerCubicMeter")]
+        public override void BaseUnit_HasSIBase()
```
Again, it's up to you to decide if we should un-`[Skip]` it or change the skip reason...

lipchev replied:
Or if you want, there is another solution that doesn't require us having the `[Skip]` warning as a permanent feature: we could use a boolean property, maybe something like `SupportsSIUnitSystem` to skip the generation of these `not-gonna-be-fixed` type of tests.. 🤔

angularsen replied:
If a unit doesn't make sense or is simply never used, then I don't think we should add it just to make some si unit system logic work, which arguably is a very niche part of UnitsNet and probably little used to begin with.

lipchev replied:
Removed the `MeterPerCubicMeter` in https://github.com/angularsen/UnitsNet/pull/1489/commits/d7169c17ec184b7a73d0817c106dd55f94b43ba7

lipchev replied:
Oh wait, it didn't remove it from the `Units.json`.. (i have to do it by hand)

lipchev commented:
I was also going to include this as part of the `UnitRelations.json`:
```  json
"Length.Kilometer = FuelEfficiency.KilometerPerLiter * Volume.Liter -- NoInferredDivision",
```
but ended up reverting the change (_not that it didn't work or anything_). Here is how this would affect the two quantities:
```  csharp
    public readonly partial struct FuelEfficiency :
        IArithmeticQuantity<FuelEfficiency, FuelEfficiencyUnit>,
#if NET7_0_OR_GREATER
        IMultiplyOperators<FuelEfficiency, Volume, Length>,
#endif
```

```  csharp
    public readonly partial struct Volume :
        IArithmeticQuantity<Volume, VolumeUnit>,
#if NET7_0_OR_GREATER
        IMultiplyOperators<Volume, Molarity, AmountOfSubstance>,
        IMultiplyOperators<Volume, ReciprocalLength, Area>,
        IDivisionOperators<Volume, Length, Area>,
        IMultiplyOperators<Volume, Length, AreaMomentOfInertia>,
        IDivisionOperators<Volume, VolumeFlow, Duration>,
        IMultiplyOperators<Volume, EnergyDensity, Energy>,
        IMultiplyOperators<Volume, FuelEfficiency, Length>,
        IMultiplyOperators<Volume, ReciprocalArea, Length>,
        IDivisionOperators<Volume, Area, Length>,
        IMultiplyOperators<Volume, Density, Mass>,
        IMultiplyOperators<Volume, MassConcentration, Mass>,
        IDivisionOperators<Volume, SpecificVolume, Mass>,
        IDivisionOperators<Volume, Mass, SpecificVolume>,
        IDivisionOperators<Volume, Duration, VolumeFlow>,
#endif
```

Here are the pros and cons to consider:
- the multiplication operator seems reasonable: we multiply some amount (volume) of fuel by some fuel efficiency and get the travel distance. Technically there isn't anything wrong with dividing some length by a given fuel efficiency, in order to get the amount of fuel- however I don't immediately think of fuel efficiency as the result of dividing a length by volume (although I can't say I have any particular _default_ quantity in mind 🤷 ).
- the downside (even with the `NoInferredDivision`) is that it _binds_ the two quantities together, making it harder (if not impossible) to separate the `Volume` and the `FuelEfficiency` later (e.g. `UnitsNet.Common` + `UnitsNet.Extended`)
- as a middle ground, we could just add (by hand) the `multiply` and `divide` operators on the side of the `FuelEfficiency`, however this might seem odd w.r.t. the multiplication..

@angularsen  A lot of important decisions for you to make in this PR 😄 

PS Of course we could just close this PR: I don't care much either way, I had the `json` file changed while testing other stuff (from months ago), so one way or another I wanted to resolve my change-set..

lipchev commented:
@angularsen  Ok, I've removed the `MeterPerCubicMeter` , skipped the tests and updated the PR summary to reflect the changes.

I've also updated the summary of the related issue: #1463, have a look at the remaining tasks..

angularsen commented:
>  the downside (even with the NoInferredDivision) is that it binds the two quantities together, making it harder (if not impossible) to separate the Volume and the FuelEfficiency later (e.g. UnitsNet.Common + UnitsNet.Extended)

I'm not sure I follow, how is this worse than binding to other quantities? Is it because you consider this a less-widely-used quantity?

Regarding extensibility and splitting up into packages, this is tricky with static typed stuff that we primarily deal with, but I have some ideas in https://github.com/angularsen/UnitsNet/pull/1500

> as a middle ground, we could just add (by hand) the multiply and divide operators on the side of the FuelEfficiency, however this might seem odd w.r.t. the multiplication..

If the inferred operators don't make sense, then yes, we always have the option of manually adding them.

lipchev commented:
> > the downside (even with the NoInferredDivision) is that it binds the two quantities together, making it harder (if not impossible) to separate the Volume and the FuelEfficiency later (e.g. UnitsNet.Common + UnitsNet.Extended)
> 
> I'm not sure I follow, how is this worse than binding to other quantities? Is it because you consider this a less-widely-used quantity?

Yes, I'd say that the _minimal quantity set_ (e.g. `UnitsNet.Common`) would have to include the SI quantities + all of their associated operators (e.g. Mass -> Density) + all of their associations (e.g. Density -> KinematicViscosity, DynamicViscosity etc..).. I haven't tested how many quantities would get covered by this graph, but If there are enough left-overs we could put them in a separate nuget.

angularsen commented:
Alright, I fear it will be tough to split up into two packages where one cannot depend on the other, but maye there is such a cut.
